### PR TITLE
Add working-set and page-fault metrics

### PR DESCRIPTION
**Pull request by pmcollins**
**Description:** Adds memory working set, page faults, and major page faults metrics to kubeletstats receiver.

**Testing:** Unit tests added using test fixtures recorded from minikube.

### CODE REVIEW

File: receiver/kubeletstatsreceiver/kubelet/mem.go [hunk 1]
```diff
@@ -38,3 +41,15 @@ func memUsageMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
...
+func memWorkingSetMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
+	return intGauge(prefix+"memory.workingset", "By", s.WorkingSetBytes)
+}
+
+func memPageFaultsMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
+	return intGauge(prefix+"memory.pagefaults", "1", s.PageFaults)
+}
+
+func memMajorPageFaultsMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
+	return intGauge(prefix+"memory.majorpagefaults", "1", s.MajorPageFaults)
```
Comment (dmitryax) [inline receiver/kubeletstatsreceiver/kubelet/mem.go:54]: I think we should make more readable: `memory.major_page_faults`. Here is an example in conventions https://github.com/open-telemetry/oteps/blob/master/text/0119-standard-system-metrics.md#systemswap

File: receiver/kubeletstatsreceiver/kubelet/mem.go [hunk 2]
```diff
@@ -38,3 +41,15 @@ func memUsageMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
 func memRssMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
 	return intGauge(prefix+"memory.rss", "By", s.RSSBytes)
 }
+
+func memWorkingSetMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
+	return intGauge(prefix+"memory.workingset", "By", s.WorkingSetBytes)
+}
+
+func memPageFaultsMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
+	return intGauge(prefix+"memory.pagefaults", "1", s.PageFaults)
```
Comment (dmitryax) [inline receiver/kubeletstatsreceiver/kubelet/mem.go:50]: same here: `memory.page_faults`

File: receiver/kubeletstatsreceiver/kubelet/mem.go [hunk 3]
```diff
@@ -38,3 +41,15 @@ func memUsageMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
 func memRssMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
 	return intGauge(prefix+"memory.rss", "By", s.RSSBytes)
 }
+
+func memWorkingSetMetric(prefix string, s *stats.MemoryStats) *metricspb.Metric {
+	return intGauge(prefix+"memory.workingset", "By", s.WorkingSetBytes)
```
Comment (dmitryax) [inline receiver/kubeletstatsreceiver/kubelet/mem.go:46]: And for consistency with others we probably need `memory.working_set` here

**Comment by dmitryax** [comment]
Could you please add signalfx translations after to:
```
container_memory_major_page_faults
container_memory_page_faults
container_memory_working_set_bytes
```
It can be done as a separate PR

**Comment by dmitryax** [comment]
Please add _kubeletstats receiver_ to the commit message

**Comment by pmcollins** [comment]
> Please add _kubeletstats receiver_ to the commit message

Done.

File: receiver/kubeletstatsreceiver/kubelet/mem.go (hunk 1 above)
Comment (pmcollins) [reply receiver/kubeletstatsreceiver/kubelet/mem.go]: Agree, fixed.

File: receiver/kubeletstatsreceiver/kubelet/mem.go (hunk 2 above)
Comment (pmcollins) [reply receiver/kubeletstatsreceiver/kubelet/mem.go]: Fixed.

File: receiver/kubeletstatsreceiver/kubelet/mem.go (hunk 3 above)
Comment (pmcollins) [reply receiver/kubeletstatsreceiver/kubelet/mem.go]: Fixed.

**Comment by pmcollins** [comment]
> Could you please add signalfx translations after to:
> 
> ```
> container_memory_major_page_faults
> container_memory_page_faults
> container_memory_working_set_bytes
> ```
> 
> It can be done as a separate PR

Separate PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/679

**Review by dmitryax — APPROVED** [review]

**Pull request merged**